src/equations/linear_scalar_advection_3d.jl

```diff
@@ -82,10 +84,12 @@ A sine wave in the conserved variable.
 """
 function initial_condition_sin(x, t, equation::LinearScalarAdvectionEquation3D)
     # Store translated coordinate for easy use of exact solution
+    RealT = eltype(x)
     x_trans = x - equation.advection_velocity * t
 
-    scalar = sin(2 * pi * x_trans[1]) * sin(2 * pi * x_trans[2]) *
-             sin(2 * pi * x_trans[3])
+    scalar = sin(2 * convert(RealT, pi) * x_trans[1]) *
+             sin(2 * convert(RealT, pi) * x_trans[2]) *
+             sin(2 * convert(RealT, pi) * x_trans[3])
```

Comment:
```suggestion
    scalar = sinpi(2 * x_trans[1]) * sinpi(2 * x_trans[2]) * sinpi(2 * x_trans[3])
```

This is a bit more brief and more exact for special values.

Reply:
Good.